src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,156 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "login";
+    private static final String VALID_PASSWORD = "password";
+    private static final String SHORT_PASSWORD = "pasik";
+    private static final Integer VALID_AGE = 18;
+    private static final Integer INVALID_AGE = 10;
+    private static final String EMPTY_LOGIN = "";
+
+    private static final User userIsNull = null;
+    private static User validUser;
+    private static User userInvalidAge;
+    private static User userNullLogin;
+    private static User userNullPassword;
+    private static User userNullAge;
+    private static User userEmptyLogin;
+    private static User userShortPassword;
+    private static RegistrationService registrationService;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        validUser = new User();
+        validUser.setLogin(VALID_LOGIN);
+        validUser.setPassword(VALID_PASSWORD);
+        validUser.setAge(VALID_AGE);
+        userInvalidAge = new User();
+        userInvalidAge.setLogin(VALID_LOGIN);
+        userInvalidAge.setPassword(VALID_PASSWORD);
+        userInvalidAge.setAge(INVALID_AGE);
+        userNullAge = new User();
+        userNullAge.setLogin(VALID_LOGIN);
+        userNullAge.setPassword(VALID_PASSWORD);
+        userNullAge.setAge(null);
+        userNullPassword = new User();
+        userNullPassword.setLogin(VALID_LOGIN);
+        userNullPassword.setPassword(null);
+        userNullPassword.setAge(VALID_AGE);
+        userNullLogin = new User();
+        userNullLogin.setLogin(null);
+        userNullLogin.setPassword(VALID_PASSWORD);
+        userNullLogin.setAge(VALID_AGE);
+        userEmptyLogin = new User();
+        userEmptyLogin.setLogin(EMPTY_LOGIN);
+        userEmptyLogin.setPassword(VALID_PASSWORD);
+        userEmptyLogin.setAge(VALID_AGE);
+        userShortPassword = new User();
+        userShortPassword.setLogin(VALID_LOGIN);
+        userShortPassword.setPassword(SHORT_PASSWORD);
+        userShortPassword.setAge(VALID_AGE);
```

Comment:
Please create 1 user with default OK values in beforeEach() method and then in each test change only required parameter(s) to perform test